WhenPushComesToShove/Assets/3-Scripts/Utilities/Hitboxes/HitboxManager.cs

```diff
@@ -0,0 +1,79 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class HitEvent
+{
+    public Hitbox hitbox;
+    public Hurtbox hurtbox;
+    public HitEvent(Hitbox hit, Hurtbox hurt)
+    {
+        this.hitbox = hit;
+        this.hurtbox = hurt;
+    }
+
+}
+
+public class HitboxManager : MonoBehaviour
+{
+    static public HitboxManager instance;
+    
+    private Dictionary<HitHandler, List<HitEvent>> hitEvents;
+
+    void Awake()
+    {
+        hitEvents = new Dictionary<HitHandler, List<HitEvent>>();
+        instance = this;
```

Comment:
I would throw this inside an if statement that makes sure instance doesn't already exist. Just a safety measure for singletons to ensure that 2 never exist at the same time especially once we start adding inventory and such that will persist over scenes.